src/libraries/Common/src/System/Security/Cryptography/ECDsaOpenSsl.cs

```diff
@@ -127,67 +141,82 @@ public override bool TrySignHash(ReadOnlySpan<byte> hash, Span<byte> destination
                     return false;
                 }
 
-                ReadOnlySpan<byte> derSignature = SignHash(hash, signDestination, signatureLength, key);
-                bytesWritten = AsymmetricAlgorithmHelpers.ConvertDerToIeee1363(derSignature, KeySize, destination);
-                Debug.Assert(bytesWritten == encodedSize);
+                // We need to duplicate key handle in case it's being used by multiple threads and one of them disposes it
+                using (SafeEvpPKeyHandle key = _key.Value.DuplicateHandle())
+                using (SafeEvpPKeyCtxHandle ctx = Interop.Crypto.EvpPKeyCtxCreate(key))
+                {
+                    Interop.Crypto.EvpPKeyCtxConfigureForECDSASign(ctx);
+
+                    if (!Interop.Crypto.TryEvpPKeyCtxSignatureSize(ctx, hash, out int sufficientSignatureSizeInBytes))
+                    {
+                        throw Interop.Crypto.CreateOpenSslCryptographicException();
+                    }
+
+                    Span<byte> derSignatureDestination = sufficientSignatureSizeInBytes <= SignatureStackBufSize ? stackalloc byte[sufficientSignatureSizeInBytes] : new byte[sufficientSignatureSizeInBytes];
```

Comment:
This is probably worth wrapping somehow. This line blows past 120 columns considerably.

Reply:
this has been changed considerably so I think the comment doesn't apply anymore